### PR TITLE
Gids 081 stage :development to staging

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts [hunk 1]
```diff
@@ -194,7 +201,20 @@ export class PaymentAsideComponent implements OnInit, OnChanges {
...
-            item.remarks = '';
+            let modifiedRemark;
+            let companyFirstName;
+            if (this.companyFirstName) {
+                companyFirstName = this.companyFirstName.split(' ');
+            }
+            if (item && item.name) {
+                modifiedRemark = item.name.split(' ');
+            }
+            if (modifiedRemark.length && companyFirstName) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts:212]: add nulity check for modifiedRemark

File: apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts [hunk 2]
```diff
@@ -377,18 +416,41 @@ export class PaymentAsideComponent implements OnInit, OnChanges {
...
+    }
+
+    /**
+     * To prevent zero from textbox
+     *
+     * @param {number} amount Amount
+     * @param {number} index Index of item
+     * @memberof PaymentAsideComponent
+     */
+    public preventZero(amount: number, index: number) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts:450]: return type

File: apps/web-giddh/src/app/services/apiurls/settings.integration.api.ts [hunk 3]
```diff
@@ -14,5 +14,6 @@ export const SETTINGS_INTEGRATION_API = {
     GET_GMAIL_INTEGRATION_STATUS: COMMON + 'gmail-token', // GET
     REMOVE_GMAIL_INTEGRATION: COMMON + 'gmail-token', //DELETE
     REMOVE_ICICI_REQUEST: COMMON + 'bank/deregistration?urn=:urn',
-    UPADTE_PAYMENT: COMMON + 'bank/updateDetails'
+    UPADTE_PAYMENT: COMMON + 'bank/updateDetails',
+    BANK_INTERATION_VALIDATION_FORM: 'ui/bank/integration/form?bankName=:bankName&='
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/apiurls/settings.integration.api.ts:18]: is &= useful at end?

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 4]
```diff
@@ -714,22 +746,74 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
...
+    /**
+     * To call when max and custom limit options change
+     *
+     * @param {*} event
+     * @param {number} index
+     * @param {boolean} isUpdate
+     * @param {number} [parentIndex]
+     * @memberof SettingIntegrationComponent
+     */
+    public maxLimitOrCustomChanged(event: any, index: number, isUpdate: boolean, parentIndex?: number, ): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:796]: remove comma after last argument

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 5]
```diff
@@ -778,6 +876,9 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
                 elementRef.classList.remove('error-box');
             }
         }
+        if (this.maxLimit) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:879]: remove this useless condition

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 6]
```diff
@@ -740,6 +824,21 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
      * @returns {boolean}
      * @memberof SettingIntegrationComponent
      */
+    public checkIsMaxBankLimitSelectedField(itemList: FormArray, index: number): boolean {
+
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:828]: remove blank spaces

File: apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts (hunk 2 above)
Comment (prateek-giddh) [inline apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts]: Add return type

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html [hunk 7]
```diff
@@ -633,28 +645,159 @@ <h3 class="fs20 mrB2">ICICI Bank</h3>
...
+                    </div>
+                    <div class="clearfix">
+                        <div style="color: gray">Once it is done you can link your ICICI account/s and make payment directly through GIDDH.
+                        </div>
+                    </div>
+                </div>
+            </div>
+             Create or add bank account end -->
+            <div *ngIf="openNewRegistration && !(isPaymentAdditionSuccess$ | async)" class="box clearfix">
+                <h3 class="fs20 mrB2">ICICI Bank</h3>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html:668]: Add all the text to locale json

File: apps/web-giddh/src/app/settings/permissions/form/form.component.html [hunk 8]
```diff
@@ -1,143 +1,219 @@
...
-                        </div>
-                        <div class="pull-left lh34">
-                            <i *ngIf="!l" (click)="delRow('allowedCidrs', i, $event)"
-                                class="fa fa-trash-o cursor-pointer"></i>
-                            <i *ngIf="l" (click)="addNewRow('allowedCidrs', item, $event)"
-                                class="fa fa-plus cursor-pointer"></i>
+    <!--  -->
+    </div>
+    <div class="mb-4">
+    <p class="mb-15">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/permissions/form/form.component.html:157]: Dummy text should be wrapped in ng-template
And new labels should be added to locale

File: apps/web-giddh/src/app/settings/permissions/form/form.component.ts [hunk 9]
```diff
@@ -47,6 +46,9 @@ export class SettingPermissionFormComponent implements OnInit, OnDestroy {
     public selectedIPRange: string = 'IP Address';
     public createPermissionInProcess$: Observable<boolean>;
     public dateRangePickerValue: Date[] = [];
+    /** To open model */
+    public opened = false;
+    public show: boolean = false;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/permissions/form/form.component.ts:51]: Missing JS doc

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 10]
```diff
@@ -714,22 +746,73 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
...
         }
-        console.log('selected', this.paymentFormObj, this.registeredAccount, index);
     }
 
-    public maxLimitOrCustomChanged(event: any, index: number, isUpdate: boolean, parentIndex?: number, ): void {
-        if (event === 'max' && !isUpdate && this.checkIsMaxBankLimitSelected(this.paymentFormObj.userAmountRanges, index)) {
-            this.paymentFormObj.userAmountRanges[index].maxBankLimit = "custom";
-            this.paymentFormObj.userAmountRanges[index].amount = null;
-            this.toasty.infoToast('You can not select max bank limit more than 1');
+    /**
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:751]: Run formatter

File: apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts (hunk 2 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts]: already done in above comment

**Pull request merged**